openquake/engine/utils/tasks.py

```diff
@@ -157,10 +157,14 @@ def wrapped(*args):
         # the revoke functionality can work
         EnginePerformanceMonitor.store_task_id(job_id, tsk)
 
+        # if there is no db handler already, add one
+        if not any(isinstance(handler, logs.LogDatabaseHandler)
+                   for handler in logs.LOG.handlers):
+            logs.LOG.addHandler(logs.LogDatabaseHandler(job))
+            logs.set_level(job.log_level)
+
```

Comment:
What happens when more than one job concur on the same worker ?
Add a test to check this behaviour could be a good choice.